cmd/oci-catalog/src/providers/dockerhub.rs

```diff
@@ -105,25 +121,79 @@ impl OCICatalogSender for DockerHubAPI {
         }
     }
 
-    async fn send_tags(&self, tx: mpsc::Sender<Result<Tag, Status>>, _request: &ListTagsRequest) {
-        for count in 0..10 {
-            tx.send(Ok(Tag {
-                name: format!("tag-{}", count),
-            }))
-            .await
-            .unwrap();
+    async fn send_tags(&self, tx: mpsc::Sender<Result<Tag, Status>>, request: &ListTagsRequest) {
```

Comment:
For what is worth, (I guess you already know it, but just in case), here is the "official" (albeit tagged as experimental) DockerHub client: https://github.com/docker/hub-tool/blob/main/pkg/hub/tags.go#L61

Reply:
Yeah, as you can see in their (go) client, they need to collect all the page results (appending to the array) before returning the result. In our case, we're wanting to stream the results back (via gRPC stream) while they're collected, which is why I'm using these `send_tags` and `send_repositories`, to show the results down a channel (which we can buffer as needed) and continue fetching.

BTW: interesting to see they use the same [default page size](https://github.com/docker/hub-tool/blob/main/pkg/hub/client.go#L43) :)

Reply:
Neat! Thanks for the explanation!